boxes/token/src/contracts/src/types/balances_map.nr

```diff
@@ -4,25 +4,32 @@ use crate::types::balance_set::BalanceSet;
 use dep::aztec::hash::pedersen_hash;
 use dep::protocol_types::address::AztecAddress;
 
+use crate::types::token_note::{TokenNote, TOKEN_NOTE_LEN, TokenNoteMethods};
+use dep::aztec::state_vars::{map::Map, set::Set};
+
 struct BalancesMap {
```

Comment:
It seems to me that after these changes it makes sense to completely nuke this struct since the reason for its existence was to have the typed key. (and maybe also the BalancesSet?) @LHerskind do you agree?

But I think it would be ok to do that in a separate PR.

Reply:
Nuking this one at last seems fair. The BalancesSet should probably be kept as is atm as it also ensures the logic around overflowing etc.

Reply:
I kept it because `owner` is used inside `BalanceSet` and keeping that additional level of indirection looked better to me than just passing `owner` around...but I really don't know which option is more idiomatic, so let me know what you guys prefer 😄

Reply:
Just to summarize, this is the problem, right?

[This](https://github.com/AztecProtocol/aztec-packages/blob/a2b267bf4af666a1848c428c2e6430fe45875bbc/boxes/token/src/contracts/src/types/balance_set.nr#L78) is the problematic line and f we didn't have there the owner then we would have to do this:
        storage.balances.at(to).add(amount, to);

Instead of:
        storage.balances.at(to).add(amount);

and we need the owner in the add function to be able to create a TokenNote which uses that info to fetch the secrets which are used to emit the logs.

I feel like we should try to use the default types in the reference Token contract but if we fail to make it not-ugly then we probably have a bigger issue here.

Reply:
Exactly right, that's the issue I see. Might be overthinking though, so open to suggestions.

Reply:
I still like the idea of removing lots of our extra layers of abstraction for our example token contract, so that people can really see an example of the raw types (of aztec.nr) being used. I.e. direct interaction with `Set`, `insert`, `remove`, emitting events etc. It would result in much more verbose function bodies. But the number of nested files to jump through to understand what's going on might be a learning hurdle for some external devs.
But I recall that wasn't a popular view last year.

Reply:
Since there's a little bit of debate around this, I'm gonna go ahead and merge this, since it still slightly simplifies the implementations. The examples are going to be revisited soon anyways with the addition of generic Traits, so there will be plenty of opportunities to improve them 😄